x/wasm-storage/types/wasm.go

```diff
@@ -51,13 +51,11 @@ func WasmTypeFromString(s string) WasmType {
 	switch strings.ToUpper(s) {
 	case "DATA-REQUEST":
 		return WasmTypeDataRequest
-	case "TALLY":
-		return WasmTypeTally
 	case "DATA-REQUEST-EXECUTOR":
 		return WasmTypeDataRequestExecutor
 	case "RELAYER":
 		return WasmTypeRelayer
 	default:
-		return WasmTypeNil
+		panic("unexpected wasm type")
```

Comment:
I guess this is safe as we're only using it in the CLI? Is there some way we can enforce that so we don't run into issues if for whatever reason later down the line we need this functionality in the chain and whoever implements it doesn't realise the function panics.

Reply:
Yes, and it's ok to use it in txs because they're isolated. You're right that we should be careful when it comes to non-tx-initiated application logic though.

I don't think there is a way to enforce that at the code level, but we can look into some code analysis engines. It is mentioned in this issue, but I will elaborate more https://github.com/sedaprotocol/seda-chain/issues/291#issue-2376044228